14/umbraco-cms/fundamentals/code/subscribing-to-notifications.md

```diff
@@ -99,29 +123,42 @@ public void Handle(ContentPublishedNotification notification)
 }
 ```
 
-Now we have a `NotificationHandler` that logs the name of a piece of content every time it's published, however, we're not done yet.
+{% endcode %}
 
-Umbraco needs to know that our handler exists and that it handles `ContentPublishedNotification`. We will do this by registering it as an extension method in the `Program.cs` file.
+## Register the Notification Handler
+
+Umbraco needs to know that our handler exists and that it handles `ContentPublishedNotification`. We need to register it in the **Program.cs** file.
 
 {% hint style="info" %}
 Registering dependencies and extensions like this can be done using different methods. Which method to use in each situation depends on whether the extension is added to the Umbraco site or a package.
 
 Learn more about registering dependencies in the [Dependency Injection](../../reference/using-ioc.md) article.
 {% endhint %}
 
-Open the `Program.cs` file at the root of the project. First, add `using Umbraco.Cms.Core.Notifications;` to the top of the file. Then, add `.AddNotificationHandler<ContentPublishedNotification, LogWhenPublishedHandler>()` right before the `Build()` part. The method now looks like this:
+1. Open the **Program.cs** file at the root of the project.
+2. Add the `using Umbraco.Cms.Core.Notifications;` statement.
 
-```csharp
-builder.CreateUmbracoBuilder()
-    .AddBackOffice()
-    .AddWebsite()
-    .AddDeliveryApi()
-    .AddComposers()
-    .AddNotificationHandler<ContentPublishedNotification, LogWhenPublishedHandler>()
-    .Build();
-```
+    ```csharp
+    using Umbraco.Cms.Core.Notifications;
+    ```
+
+3. Register the handler in the builder configuration by adding the `.AddNotificationHandler<ContentPublishedNotification, LogWhenPublishedHandler>()` method call.
+
+    The registration should look like this:
+
+    ```csharp
+    builder.CreateUmbracoBuilder()
+        .AddBackOffice()
+        .AddWebsite()
+        .AddDeliveryApi()
+        .AddComposers()
+        .AddNotificationHandler<ContentPublishedNotification, LogWhenPublishedHandler>()
+        .Build();
+    ```
 
-The entire handler class should look like this:
+    The entire handler class should look like this:
```

Comment:
I'm not sure it makes sense to add the full LogWhenPublishedHandler.cs file here. It was a little confusing, as it comes after we need to add something to the PRogram.cs file, which make it sound like the code snippet is actually how the full program.cs should look 🙈

Reply:
Makes sense. I'll add it as a dropdown.